test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/architecture/AbstractCloudArchitectureIntegrationTest.java

```diff
@@ -0,0 +1,188 @@
+import org.kie.cloud.api.scenario.GenericScenario;
+import org.kie.cloud.integrationtests.AbstractCloudIntegrationTest;
+import org.kie.server.api.exception.KieServicesHttpException;
+import org.kie.server.api.model.KieContainerResource;
+import org.kie.server.api.model.KieContainerResourceList;
+import org.kie.server.api.model.KieContainerStatus;
+import org.kie.server.api.model.KieServerInfo;
+import org.kie.server.api.model.ServiceResponse;
+import org.kie.server.client.KieServicesClient;
+import org.kie.server.client.QueryServicesClient;
+import org.kie.server.controller.api.model.runtime.ServerInstanceKey;
+import org.kie.server.controller.api.model.spec.ContainerSpec;
+import org.kie.server.integrationtests.controller.client.KieServerMgmtControllerClient;
+import org.kie.server.integrationtests.router.client.KieServerRouterClient;
+import org.kie.server.router.Configuration;
+
+public abstract class AbstractCloudArchitectureIntegrationTest extends AbstractCloudIntegrationTest<GenericScenario> {
+
+    protected static final String SMART_ROUTER_ID = "test-kie-router";
+
+    protected static final String
+            CONTAINER_ID_ABC = CONTAINER_ID + "-abc",
+            CONTAINER_ID_DEF = CONTAINER_ID + "-def",
+            CONTAINER_ID_GHI = CONTAINER_ID + "-ghi";
+    protected static final String
+            KIE_CONTAINER_DEPLOYMENT_ABC = CONTAINER_ID_ABC + "=" + PROJECT_GROUP_ID + ":" + DEFINITION_PROJECT_NAME + ":" + DEFINITION_PROJECT_VERSION,
+            KIE_CONTAINER_DEPLOYMENT_DEF = CONTAINER_ID_DEF + "=" + PROJECT_GROUP_ID + ":" + DEFINITION_PROJECT_NAME + ":" + DEFINITION_PROJECT_VERSION,
+            KIE_CONTAINER_DEPLOYMENT_GHI = CONTAINER_ID_GHI + "=" + PROJECT_GROUP_ID + ":" + DEFINITION_PROJECT_NAME + ":" + DEFINITION_PROJECT_VERSION;
+
+    protected static final String REPO_BRANCH = "master";
+
+    protected static final String CONTROLLER_NAME = "controller";
+    protected static final String SMART_ROUTER_NAME = "smart-router";
+
+    protected static final String
+            KIE_SERVER_ABC_NAME = "kie-server-abc",
+            KIE_SERVER_DEF_NAME = "kie-server-def",
+            KIE_SERVER_GHI_NAME = "kie-server-ghi";
+
+    protected static final String PROJECT_SOURCE_FOLDER = "/kjars-sources";
+
+    protected static final String PORT = "80";
+
+    protected String repositoryName;
+
+    protected String getKieServerId(KieServicesClient kieServerClient) {
+        ServiceResponse<KieServerInfo> serverInfo = kieServerClient.getServerInfo();
+        assertThat(serverInfo.getType()).isEqualTo(ServiceResponse.ResponseType.SUCCESS);
+
+        return serverInfo.getResult().getServerId();
+    }
+
+    protected void verifyClientKieServersLocationCount(KieServicesClient kieServerClient, final int numberOfExpectedServers) {
+        if (numberOfExpectedServers == 0) {
+            ServiceResponse<KieServerInfo> serverInfo;
+            try {
+                serverInfo = kieServerClient.getServerInfo();
```

Comment:
Personally, I'd fit these assertions into the try-catch block. Perhaps to its own method, `assertOnKieServerInfo()`.